src/fsharp/CompileOps.fs

```diff
@@ -4888,9 +4902,128 @@ module private ScriptPreprocessClosure =
+    let PM_COMMAND = "install --generate-load-scripts"
+    let PM_LOADSCRIPT = PM_DIR + "/load/main.group.fsx"
+
+    let userProfile = 
+        let res = Environment.GetEnvironmentVariable("USERPROFILE")
+        if System.String.IsNullOrEmpty res then
+            Environment.GetEnvironmentVariable("HOME")
+        else res
+
+    let ResolvePackages (implicitIncludeDir: string, scriptName: string, packageManagerTextLines: string list, m) =
+        printfn "OBJ %A" (implicitIncludeDir,scriptName)
+        let workingDir = Path.Combine(Path.GetTempPath(),"fsx-packages", string(abs(hash (implicitIncludeDir,scriptName))))
+        let workingDirSpecFile = FileInfo(Path.Combine(workingDir,PM_SPEC_FILE))
+        if not (Directory.Exists workingDir) then
+            Directory.CreateDirectory workingDir |> ignore
+
+        let packageManagerTextLines = packageManagerTextLines |> List.filter (fun l -> not (String.IsNullOrWhiteSpace l))
+
+        let rootDir,packageManagerTextLines =
+            let rec findSpecFile dir =
+                let fi = FileInfo(Path.Combine(dir,PM_SPEC_FILE))
+                if fi.Exists then
+                    let lockFile = FileInfo(Path.Combine(fi.Directory.FullName,PM_LOCK_FILE))
+                    let depsFileLines = File.ReadAllLines fi.FullName
+                    if lockFile.Exists then
+                        let originalDepsFile = FileInfo(workingDirSpecFile.FullName + ".original")
+                        if not originalDepsFile.Exists ||
+                           File.ReadAllLines originalDepsFile.FullName <> depsFileLines
+                        then
+                            File.Copy(fi.FullName,originalDepsFile.FullName,true)
+                            let targetLockFile = FileInfo(Path.Combine(workingDir,PM_LOCK_FILE))
+                            File.Copy(lockFile.FullName,targetLockFile.FullName,true)
+                        
+                    fi.Directory.FullName, (Array.toList depsFileLines) @ packageManagerTextLines
+                elif fi.Directory.Parent <> null then
+                    findSpecFile fi.Directory.Parent.FullName
+                else
+                    workingDir, "framework: net461" :: "source https://nuget.org/api/v2" :: packageManagerTextLines
+           
+            findSpecFile implicitIncludeDir
+
+        let toolPathOpt = 
+            let rec loop dir = 
+                if Directory.Exists dir then 
+                    let dirExe = Path.Combine (dir, PM_EXE)
+                    if File.Exists dirExe then Some dirExe else
+                    let paketDir = Path.Combine (dir, PM_DIR)
+                    let paketDirExe = Path.Combine (paketDir, PM_EXE)
+                    if Directory.Exists paketDir && File.Exists paketDirExe then Some paketDirExe
+                    elif not (String.IsNullOrWhiteSpace (Path.GetDirectoryName(paketDir))) then 
+                        loop (Path.GetDirectoryName(paketDir))
```

Comment:
fix + split in additional file to help with scriptcs integration in https://github.com/forki/visualfsharp/pull/14